ios/SubscriptionsDB.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  SubscriptionsSQLite.swift
+//  magnet
+//
+//  Created by Wilson Page on 26/10/2016.
+//
+
+import Foundation
+import SQLite
+
+class SubscriptionsDB {
+  private static var singleton: SubscriptionsDB!
+  private let table: Table = Table("subscriptions")
+  private let columnChannelName = Expression<String>(SubscriptionRecord.CHANNEL_NAME);
+  private let columnNotificationsEnabled = Expression<Bool>(SubscriptionRecord.NOTIFICATIONS_ENABLED)
+  private var db: Connection
+
+  static func get() -> SubscriptionsDB {
+    let lockQueue = dispatch_queue_create("com.mozilla.magnet.subscriptions", nil)
+    
+    // syncronized
+    dispatch_sync(lockQueue) {
+      if (singleton == nil) {
+        singleton = SubscriptionsDB()
+      }
+    }
+    
+    return singleton
+  }
```

Comment:
Moved the singleton factory in to the SQL class so that I could extend the `Subscriptions` class to make `SubscriptionsReact`.